src/inject_typed/container.py

```diff
@@ -1,18 +1,75 @@
-from typing import Type, Any, TypeVar
+from typing import Type, Any, TypeVar, Dict
+from inspect import signature, _empty
 
 T1 = TypeVar('T1', covariant=True)
 
 
 class ContainerException(Exception):
     pass
 
+
+_IGNORED_PARAMS = {'args', 'kwargs', 'self'}
+
+
+def _class_dependencies(clazz: Type[Any]) -> Dict[str, Any]:
+    constructor_signature = signature(clazz.__init__)
+    filtered_params = {key: param for (key, param) in constructor_signature.parameters.items() if
+                       key not in _IGNORED_PARAMS}
+    return filtered_params
+
+
+def _fulfills_dependency(clazz: Type[Any], dependency: Type[Any]) -> bool:
+    return clazz == dependency or dependency in clazz.__bases__
+
+
 class Container:
 
+    def __init__(self):
+        self._classes_dependencies = {}
+        self._bindings = {}
+
     def add_class(self, clazz: Type[Any]) -> None:
-        raise NotImplementedError()
+        self._classes_dependencies[clazz] = _class_dependencies(clazz)
+
+    def bind_value(self, clazz: Type[T1], value: T1) -> None:
+        self._classes_dependencies[clazz] = {}
+        self._bindings[clazz] = value
+
+    def get(self, clazz: Type[T1]) -> T1:
+        self._perform_checks(clazz)
+        if clazz not in self._bindings.keys():
+            self._create_instance(clazz)
+
+        return self._bindings[clazz]
+
+    def _create_instance(self, clazz: Type[Any]) -> None:
+        dependencies = self._get_dependencies(clazz)
+        self._bindings[clazz] = clazz(**dependencies)
+
+    def _get_dependencies(self, clazz) -> Dict[str, Any]:
+        return {key: self.get(param.annotation) for (key, param) in self._classes_dependencies[clazz].items()}
+
+    def _perform_checks(self, clazz: Type[Any]) -> None:
```

Comment:
Is this check recursive?
Would it work on these two classes?

```
class Foo
	def __init__(self, bar: Bar)


class Bar
	def __init__(self, foo: Foo)
```

My guess is it would loop ad infinitum